rinad/src/ipcm/ipcm.cc

```diff
@@ -1151,6 +1154,69 @@ IPCManager_::select_policy_set(Addon* callee, Promise* promise,
 	return IPCM_PENDING;
 }
 
+// Returns true if a kernel plugin was successfully loaded/unloaded
+ipcm_res_t
+IPCManager_::plugin_load_kernel(Addon* callee, Promise* promise,
+				const std::string& plugin_name, bool load)
+{
+	TransactionState* trans;
+	ostringstream ss;
+	ipcm_res_t result = IPCM_FAILURE;
+	pid_t pid;
+
+	trans = new TransactionState(callee, promise);
+	if (!trans){
+		ss << "Unable to allocate memory for the transaction object. Out of memory! ";
+		FLUSH_LOG(ERR, ss);
+		throw rina::Exception();
+	}
+
+	// Store transaction
+	if (add_transaction_state(trans) < 0){
+		ss << "Unable to add transaction; out of memory? ";
+		FLUSH_LOG(ERR, ss);
+		throw rina::Exception();
+	}
+
+	pid = fork();
+	if (pid < 0) {
+		// parent, fork() failed
+		ss << "Kernel plugin (un)loading: fork() failed";
+		FLUSH_LOG(ERR, ss);
+		result = IPCM_FAILURE;
+
+	} else if (pid == 0) {
+		// child
+		if (load) {
+			execl("modprobe", "modprobe", plugin_name.c_str(), NULL);
+			execlp("modprobe", "modprobe", plugin_name.c_str(), NULL);
+
+		} else {
+			execl("rmmod", "rmmod", plugin_name.c_str(), NULL);
```

Comment:
Well, modprobe handles module dependencies as well. modprobe -r: If the modules it depends on are also unused, modprobe will try to remove them too.

I always see it as:
insmod / rmmod
modprobe / modprobe -r